engine/client/secretprovider/env.go

```diff
@@ -0,0 +1,21 @@
+package secretprovider
+
+import (
+	"context"
+	"fmt"
+	"os"
+)
+
+func envProvider(_ context.Context, name string) ([]byte, error) {
+	v, ok := os.LookupEnv(name)
+	if !ok {
+		// Don't show the entire env var name, in case the user accidentally passed the value instead...
+		// This is important because users originally *did* have to pass the value, before we changed to
+		// passing by name instead.
```

Comment:
nit: I'd personally be fine rm'ing this cruft now that it's been a while since we supported providing plaintext directly